openhands/storage/conversation/file_conversation_store.py

```diff
@@ -83,13 +97,41 @@ async def search(
         start = page_id_to_offset(page_id)
         end = min(limit + start, num_conversations)
         conversations = []
+        orphaned_dirs = []  # Track directories that should be cleaned up
+
         for conversation_id in conversation_ids:
             try:
                 conversations.append(await self.get_metadata(conversation_id))
-            except Exception:
+            except FileNotFoundError:
+                # This is likely a race condition where the directory exists but metadata.json is missing
+                # This happens when delete_conversation removes the directory but other operations
+                # recreate it without the metadata.json file
+                logger.debug(
```

Comment:
If we're encountering race conditions a lot that's worrisome, so I think we should at least issue a warning.
```suggestion
                logger.warning(
```